pkgs/by-name/sn/snx-rs/package.nix

```
@@ -1,51 +1,74 @@
 {
   fetchFromGitHub,
-  rustPlatform,
-  lib,
-  pkg-config,
-  openssl,
   glib,
-  atk,
   gtk3,
+  iproute2,
+  kdePackages,
+  lib,
+  libappindicator,
+  libappindicator-gtk2,
+  libappindicator-gtk3,
+  libayatana-appindicator,
   libsoup_3,
+  openssl,
+  pkg-config,
+  rustPlatform,
   webkitgtk_4_1,
+  nix-update-script,
 }:
-rustPlatform.buildRustPackage {
+rustPlatform.buildRustPackage rec {
   pname = "snx-rs";
-  version = "2.2.3";
+  version = "2.9.0";
 
   src = fetchFromGitHub {
     owner = "ancwrd1";
     repo = "snx-rs";
-    rev = "v2.2.3";
-    hash = "sha256-tBl67uDeYVmVBwi8NQVclfFQ0Sj1dl+hR8Jct1iE2LI=";
+    rev = "v2.9.0";
+    hash = "sha256-9uTawt9Lk3YJbw8hzrP91bEaPGT1wZehrmCWDKYrw5w=";
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  passthru.updateScript = nix-update-script { };
+
+  nativeBuildInputs = [
+    iproute2
+    pkg-config
+  ];
+
   buildInputs = [
-    openssl
     glib
-    atk
     gtk3
+    kdePackages.kstatusnotifieritem
+    libappindicator
+    libappindicator-gtk2
+    libappindicator-gtk3
+    libayatana-appindicator
     libsoup_3
+    openssl
     webkitgtk_4_1
   ];
 
+  postPatch = ''
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
```

```suggestion
      --replace-fail "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
```
Maybe use `replace-fail` to warn us when something goes awry?

---

Got it. Updating the branch. Another little thing learned on how Nix works.

---

The latest push adds what you suggested.